packages/@webex/internal-plugin-metrics/src/new-metrics.ts

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable valid-jsdoc */
+
+// @ts-ignore
+import {WebexPlugin} from '@webex/webex-core';
+
+import CallDiagnosticMetrics, {
+  SubmitClientEventOptions,
+} from './call-diagnostic/call-diagnostic-metrics';
+import BehavioralMetrics from './behavioral/behavioral-metrics';
+import {
+  RecursivePartial,
+  ClientEvent,
+  FeatureEvent,
+  BehavioralEvent,
+  OperationalEvent,
+  MediaQualityEvent,
+} from './types';
+import CallAnalyzerLatencies from './call-diagnostic/call-diagnostic-metrics-latencies';
+
+/**
+ * Metrics plugin to centralize all types of metrics.
+ */
+class Metrics extends WebexPlugin {
+  // eslint-disable-next-line no-use-before-define
+  static instance: Metrics;
+
+  // latencies required for CA
+  callAnalyzerLatencies: CallAnalyzerLatencies;
+
+  // Helper classes to handle the different types of metrics
+  callDiagnosticMetrics: CallDiagnosticMetrics;
+  behavioralMetrics: BehavioralMetrics;
+
+  /**
+   *
+   * @param args
+   * @returns
+   */
+  constructor(...args) {
+    super(...args);
+
+    if (!Metrics.instance) {
+      Metrics.instance = this;
+    }
+
+    this.callAnalyzerLatencies = new CallAnalyzerLatencies();
+
+    this.callDiagnosticMetrics = new CallDiagnosticMetrics();
+    this.behavioralMetrics = new BehavioralMetrics();
+
+    // eslint-disable-next-line no-constructor-return
+    return Metrics.instance;
+  }
+
+  /**
+   * Initialize Call diagnostic class
+   * @param meetingCollection
+   * @param webex
+   */
+  initialSetupCallDiagnosticMetrics(meetingCollection: any, webex: object) {
```

Comment:
Since this Metrics class extends WebexPlugin, I'm not entirely sure if it's ok to consider it as a Singleton?
Also, at the moment i'm storing webex object and meetingCollection only on the `CallDiagnosticMetrics` class, but I'm thinking wether it makes sense now.. since this (Metrics) class is a WebexPlugin, maybe I already have access to both?


Not sure how these 2 things work, so will need some imput on this one.

Reply:
CC: @maxinteger @marcin-bazyl @arun3528

Reply:
you do not export the `Metrics` class only one instance, that should cover the Singleton  part.

Yes, you can use `this.webex` in `Metrics`, but probably better to store it in `CallDiagnosticMetrics` as well